onnxruntime/core/platform/linux/device_discovery.cc

```diff
@@ -114,6 +115,27 @@ std::optional<bool> IsGpuDiscrete(uint16_t vendor_id, uint16_t device_id) {
   return std::nullopt;
 }
 
+constexpr static const char* VendorIdToString(OrtDevice::VendorId vendorId) {
```

Comment:
this is implicitly depending on `include/onnxruntime/core/framework/ortdevice.h` which is not ideal (see this [comment](https://github.com/microsoft/onnxruntime/pull/26210#discussion_r2450054530)). is setting the vendor name necessary for this PR? if not, maybe we can refine the vendor ID/name setup in another PR.